pkgs/tools/networking/zap/default.nix

```diff
@@ -23,7 +23,7 @@ stdenv.mkDerivation rec {
     #!${runtimeShell}
     export PATH="${lib.makeBinPath [ jre ]}:\$PATH"
     export JAVA_HOME='${jre}'
-    if ! [ -f "~/.ZAP/config.xml" ];then
+    if ! [ -f "${HOME}/.ZAP/config.xml" ];then
```

Comment:
Are you sure the works? As I can't see a binding `HOME` in scope, you probably want `''${HOME}` or `$HOME` to get the shell variable expansion in the shell script, rather than having nix using interpolation to eagerly interpolate a string.

Though looking at it more closer, you might even need to add another backslash to avoid that being expanded already early…

Perhaps it is even better to move this script into a `writeShellScript` and `cp`/`ln -s` that, rather than using a `cat` to file… We are piling to many layers of expansion and escaping and it gets tedious to understand which layer we are in or want to let the expansion happen.

----

PS: had some free ressources and indeed eval fails:

```text
$ nix build github:nixos/nixpkgs/pull/219740/head#zap
error:
       … while evaluating a branch condition

         at /nix/store/ynxgyfdbm5vnjgalrv61ym8ridhk1pz1-source/lib/customisation.nix:171:8:

          170|
          171|     in if missingArgs == [] then makeOverridable f allArgs else throw error;
             |        ^
          172|

       … while calling the 'attrNames' builtin

         at /nix/store/ynxgyfdbm5vnjgalrv61ym8ridhk1pz1-source/lib/customisation.nix:131:21:

          130|       # wouldn't be passed to it
          131|       missingArgs = lib.attrNames
             |                     ^
          132|         # Filter out arguments that have a default value

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: undefined variable 'HOME'

       at /nix/store/ynxgyfdbm5vnjgalrv61ym8ridhk1pz1-source/pkgs/tools/networking/zap/default.nix:26:18:

           25|     export JAVA_HOME='${jre}'
           26|     if ! [ -f "${HOME}/.ZAP/config.xml" ];then
             |                  ^
           27|       mkdir -p "\$HOME/.ZAP"
```